### DISCUSSION TITLE
High Level Comparison vs Data-Lakehouse Technologies

### DISCUSSION THREAD

**Discussion by haydenflinner** [apache/doris#16682] [A - General / Q&A]
Let's take as an example the "Minio(S3) + Nessie + Iceberg + Flink" "lakehouse" stack; a combo of off-the-shelf tools attempting to do the same thing as Doris, but cloud-native (read: object-store). In order: Minio is providing an obejct storage abstraction over disk. Nessie is providing a catalog of where to find Iceberg table info, i.e. trying to stay out of the way. Iceberg is a format for storing files on s3 such that query engines can easily pushdown predicates to the data files, + transactions. Then Flink is query engine, asking Nessie where to find Iceberg tables on S3, and/or writing them.

How does doris compare? Certainly it is far easier to get started, and probably much faster if my experience with Iceberg-Spark writes is any comparison. But which parts of doris am I stuck with vs which are plug-and-play?

On storage for example, I see Iceberg support is being added to Doris, and I see some note on which types of deletes are supported. **Does this mean that inserts to Iceberg are supported**? If inserts and multiple catalogs are supported, I guess I can use Doris to write Iceberg files to S3? Then I am using Doris basically as a query engine, like Flink (but probably far faster 😄 ). 

I also have the option to write Doris-native storage to my local disk. If Iceberg isn't yet a first-class-citizen, _must_ I use local-disk writes? And then the Doris cluster handles replication across cluster's disks? I guess by 'end-to-end principle', this is probably faster than using network-attached storage (no matter how performant) and then setting Doris replication level to 1. But I can't for example give Doris the credentials to an s3 bucket (which is actually running locally, so not slow), and let it use that rather than POSIX filesystem?

But if I want to add more compute without adding more storage or downloading a copy of the data, how does that work? I found this page in the process of drafting this question, but it seems early days for this use-case.
https://doris.apache.org/docs/dev/advanced/compute_node/#scenario
https://github.com/apache/doris/issues/13144

**Comment by morningman** [comment]
Hi @haydenflinner ,
The lakehouse feature in Doris is still under devolepment, as you can see in [Doris Roadmap 2023](https://github.com/apache/doris/issues/16392), the Lakehouse & Data Integration part.

> On storage for example, I see Iceberg support is being added to Doris, and I see some note on which types of deletes are supported. Does this mean that inserts to Iceberg are supported? If inserts and multiple catalogs are supported, I guess I can use Doris to write Iceberg files to S3? Then I am using Doris basically as a query engine, like Flink (but probably far faster 😄 ).

For now, the iceberg data is managed(write-and-commit) beyond the Doris.
As you said, user have to use other engines(spark/flink) to insert data to iceberg, and use Doris to query.
But we are working on "self-managed lake engine", which can query and insert data to iceberg by Doris directly.

> I also have the option to write Doris-native storage to my local disk. If Iceberg isn't yet a first-class-citizen, must I use local-disk writes? And then the Doris cluster handles replication across cluster's disks? I guess by 'end-to-end principle', this is probably faster than using network-attached storage (no matter how performant) and then setting Doris replication level to 1. But I can't for example give Doris the credentials to an s3 bucket (which is actually running locally, so not slow), and let it use that rather than POSIX filesystem?

After supporting insert data to iceberg using Doris, the data will be wrote directly to s3(or other cloud storage).
And the local disk will be used as write-through cache or some kind of cache layer to accelerate query.

> But if I want to add more compute without adding more storage or downloading a copy of the data, how does that work? I found this page in the process of drafting this question, but it seems early days for this use-case.

As you see, now the data is on s3, so the BE node become stateless, we can easily add or drop compute node without worrying about data replication or migration.

> How does Doris compare?

Flink -> QueryEngine in Doris
Nessie -> Catalog in Doris (also support external catalog service such as Glue, Hive Metastore)
iceberg -> iceberg
s3 -> s3
